doozer/doozerlib/backend/konflux_fbc.py

```diff
@@ -225,6 +231,278 @@ async def _get_package_name(self, metadata: ImageMetadata) -> str:
+            yaml.dump(target_idms, ims_file)
+        logger.info("ImageDigestMirrorSet written to %s", idms_path)
+
+        # Render the FBC fragments
+        logger.info("Rendering FBC fragments...")
+        all_blobs = await asyncio.gather(*(opm.render(fragment, output_format="yaml") for fragment in fragments))
+        catalog_blobs = list(itertools.chain.from_iterable(all_blobs))
+
+        # Write the rendered blobs to a file
+        # Write catalog_blobs to catalog/<package>/catalog.yaml
+        catalog_dir = repo_dir.joinpath("catalog")
+        catalog_dir.mkdir(parents=True, exist_ok=True)
+        catalog_file_path = catalog_dir.joinpath("catalog.yaml")
+        LOGGER.info("Writing catalog blobs to %s", catalog_file_path)
+        with catalog_file_path.open('w') as f:
+            yaml.dump_all(catalog_blobs, f)
+
+        # Generate Dockerfile
+        df_path = repo_dir.joinpath("catalog.Dockerfile")
+        LOGGER.info("Generating Dockerfile %s", df_path)
+        if df_path.exists():
+            LOGGER.info("Removing existing Dockerfile %s", df_path)
+            df_path.unlink()
+        base_image_format = (
+            BASE_IMAGE_RHEL9_PULLSPEC_FORMAT if (major, minor) >= (4, 15) else BASE_IMAGE_RHEL8_PULLSPEC_FORMAT
+        )
+        base_image = base_image_format.format(major=major, minor=minor)
+        await opm.generate_dockerfile(repo_dir, "catalog", base_image=base_image, builder_image=base_image)
+
+        # Validate the catalog
+        logger.info("Validating the catalog")
+        await opm.validate(catalog_dir)
+
+        # Commit the changes to the git repository
+        if not self.dry_run:
+            logger.info("Committing changes to the git repository...")
+            message = self.commit_message or f"Merge FBC fragments\n\n{', '.join(fragments)}"
+            await build_repo.commit(message, allow_empty=True)
+            logger.info("Pushing changes to the remote repository...")
+            await build_repo.push()
+            logger.info(f"Changes pushed to remote repository {self.fbc_git_repo} branch {self.fbc_branch}")
+        else:
+            logger.info("Dry run enabled, not pushing changes.")
+
+        # Start the build
+        retries = 3
+        for attempt in range(retries):
+            logger.info(f"Attempt {attempt + 1} of {retries} to start the build...")
+            error = None
+            created_plr = await self._start_build(
+                build_repo=build_repo,
+                output_image=target_index,
+            )
+            plr_name = created_plr["metadata"]["name"]
+            plr_url = konflux_client.resource_url(created_plr)
+            logger.info(f"Created Pipelinerun {plr_name}: {plr_url}")
+
+            # Wait for the Pipelinerun to complete
+            logger.info(f"Waiting for Pipelinerun {plr_name} to complete...")
+            plr, _ = await konflux_client.wait_for_pipelinerun(plr_name, self.konflux_namespace)
```

Comment:
so this fires an fbc build? I think we'll want to store this in the db? 
And how would this need to be released?

Reply:
Yes, it fires an fbc build on Konflux. The build record is stored in the DB so far (unless it is needed in the future).

> And how would this need to be released?
It doesn't need to be released. The build is configured to push the resulting image to `quay.io/openshift-art/stage-fbc-fragments:ocp-{major}.{minor}`, which can be consumed by ERT.